prog/base.rb

```diff
@@ -179,6 +191,7 @@ def leaf?
   # A hop is a kind of jump, as in, like a jump instruction.
   def hop(label)
```

Comment:
This can probably be given a longer name and even made private. Such short names usually indicate common use, now obsoleted by `hop_methname`.

Reply:
I wasn't aware of reasoning of `unsynchronized_run` name. It's good trick.


I renamed `hop` as `dynamic_hop`, and made it private.